e2e/test/scenarios/native/native-database-source.cy.spec.js

```diff
@@ -0,0 +1,247 @@
+
+      selectDatabase(adminPersistedDatabase);
+
+      startNativeQuestion();
+      assertSelectedDatabase(adminPersistedDatabase);
+
+      cy.signOut();
+      cy.signInAsNormalUser();
+
+      startNativeQuestion();
+      assertNoDatabaseSelected();
+
+      selectDatabase(userPersistedDatabase);
+
+      startNativeQuestion();
+      assertSelectedDatabase(userPersistedDatabase);
+
+      cy.signOut();
+      cy.signInAsAdmin();
+
+      startNativeQuestion();
+      assertSelectedDatabase(adminPersistedDatabase);
+    });
+
+    it("deleting previously persisted database should result in the new database selection prompt", () => {
+      addPostgresDatabase(additionalPG);
+
+      startNativeQuestion();
+      assertNoDatabaseSelected();
+
+      selectDatabase(additionalPG);
+
+      cy.log("Delete previously persisted database.");
+      cy.get("@postgresID").then(databaseId => {
+        cy.request("DELETE", `/api/database/${databaseId}`);
+      });
+
+      startNativeQuestion();
+      assertNoDatabaseSelected();
+    });
+
+    it("persisting a database source should work between native models and questions intechangeably", () => {
+      startNativeModel();
+      assertNoDatabaseSelected();
+
+      selectDatabase("QA Postgres12");
+
+      startNativeQuestion();
+      assertSelectedDatabase("QA Postgres12").click();
+      selectDatabase("Sample Database");
+
+      startNativeModel();
+
+      cy.findByTestId("native-query-top-bar").should(
+        "not.contain",
+        "Select a database",
+      );
+      cy.findByTestId("selected-database").should(
+        "have.text",
+        "Sample Database",
+      );
+    });
+
+    it("should not update the setting when the same database is selected again", () => {
+      cy.request("PUT", "/api/setting/last-used-database-id", {
+        value: SAMPLE_DB_ID,
+      });
+
+      startNativeQuestion();
+      cy.findByTestId("selected-database")
+        .should("have.text", "Sample Database")
+        .click();
+
+      cy.log("Pick the same database again");
+      selectDatabase("Sample Database");
+      cy.get("@persistDatabase").should("be.null");
+    });
+
+    it("selecting a database in native editor for model actions should not persist the database", () => {
+      [SAMPLE_DB_ID, PG_DB_ID].forEach(enableModelActionsForDatabase);
+
+      startNewAction();
+      assertNoDatabaseSelected();
+
+      selectDatabase("Sample Database");
+      cy.get("@persistDatabase").should("be.null");
+
+      startNativeModel();
+      assertNoDatabaseSelected();
+      cy.log(
+        "Persisting a database for a native model should not affect actions",
+      );
+      selectDatabase("QA Postgres12");
+      cy.wait("@persistDatabase");
+
+      startNewAction();
+      assertNoDatabaseSelected();
+    });
+
+    describe("permissions", () => {
+      it("users with 'No self-service' data permissions should be able to choose only the databases they can query against", () => {
```

Comment:
I think this scenario (similar to when DB is removed) is also worth testing:

1. User has access to DB X
2. User creates a native query based on DB X
3. User is revoked permissions to DB X
4. User opens native query editor

Assertion: DB X is not listed and not selected

Reply:
Added a new test for this.
As with the other scenarios, unless there is a hard refresh user will still see the old database.

Spoke to AlexP about this and we concluded that this is a global problem we have - definitely not tied to just this PR.